invokeai/frontend/web/src/features/gallery/components/Boards/BoardsList/BoardEditableTitle.tsx

```diff
@@ -67,17 +77,28 @@ export const BoardEditableTitle = memo(({ board, isSelected }: Props) => {
 
   if (!isEditing.isTrue) {
     return (
-      <Text
-        size="sm"
-        fontWeight="semibold"
-        userSelect="none"
-        color={isSelected ? 'base.100' : 'base.300'}
-        onDoubleClick={isEditing.setTrue}
-        cursor="text"
-        minW={16}
-      >
-        {localTitle}
-      </Text>
+      <Flex alignItems="center" gap={1} onMouseOver={handleMouseOver} onMouseOut={handleMouseOut}>
+        <Text
+          size="sm"
+          fontWeight="semibold"
+          userSelect="none"
+          color={isSelected ? 'base.100' : 'base.300'}
+          onDoubleClick={isEditing.setTrue}
+          cursor="text"
+          minW={16}
+        >
+          {localTitle}
+        </Text>
+        {isHovering && (
+          <IconButton
+            aria-label="edit name"
+            icon={<PiPencilBold />}
+            size="sm"
+            variant="ghost"
+            onClick={isEditing.setTrue}
+          />
+        )}
```

Comment:
Why is this needed? You can click the board title to edit it.

Reply:
Also this introduces a race condition (ish) where the icon doesn't disappear.

https://github.com/user-attachments/assets/cf32e005-037e-424c-828e-2052da34147a

Reply:
Ok, I think there's just a typo in the PR description - mentions right-click but it actually means double-click, all good. I'll tidy up the minor issues I've mentioned in a followup